src/app/services/project.service.ts

```diff
@@ -48,25 +48,69 @@ export class ProjectService {
   }
 
   private getCollections() {
-    return this.api.getProjectCollections(this.project.name.replace(/\W+/g, '-').toLowerCase()).pipe(
-      map((res: any) => this.processCollections(res)),
-      map((collections: any[]) => {
+    return this.api.getProjectCollections(this.project._id).pipe(
+      map((res: any) => this.processCollections(res && res[0] && res[0].searchResults ? res[0].searchResults : null)),
+      map(async (collections: any[]) => {
         // Push them into the project
-        collections.forEach(collection => {
+        await this.loadCollectionRecords(collections);
+
+        collections.filter(c => c.documents.length > 0).forEach(collection => {
           this.addCollection(this.project.collections, collection);
         });
-      }));
+      })
+    );
   }
 
   private processCollections(res: any[]): any[] {
     const collections = res ? res : [];
 
     collections.forEach((collection, index) => {
-      collections[index] = new Collection(this.api.hostnameNRPTI, this.api.hostnameNRPTI, collection);
+      collections[index] = new Collection(collection);
     });
+
     return collections;
   }
 
+  private async loadCollectionRecords(collectionsList: any[]) {
```

Comment:
It looks like we are building the collections and with records and documents on the front.  Would it be better to make some changes to the back to just send all this back at once?

Reply:
Yeah, thought about changing it to use an endpoint that pre-populated, but I wanted to try and use existing endpoints for consistency. If there's a desire to move this to a specialized endpoint I can add that to the NRPTI api, but this will work for just BCMI if it'll be the only place it would get used.